backend/services/market_data_service_waterfall.py

```
@@ -197,6 +197,83 @@ def _update_cache(self, cache_type: str, symbol: str, result: Dict[str, Any]):
+                missing.append(symbol)
+
+        if not missing:
+            return results
+
+        # 2. Bulk Fetch YFinance
+        if self._can_use('yfinance'):
+            try:
+                self._use('yfinance')
+                import yfinance as yf
+                import math
+                import pandas as pd
```

Flake8 (7.3.0)

[error] 'pandas as pd' imported but unused

(F401)